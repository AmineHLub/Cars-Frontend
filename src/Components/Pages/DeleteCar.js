import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import axios from 'axios';
import { useNavigate } from 'react-router-dom';
import baseUrl from '../../Redux/State/baseUrl';
import { fetchCars } from '../../Redux/State/Car';

const DeleteCar = () => {
  const carsList = useSelector((state) => state.carReducer);
  const navigate = useNavigate();

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCars());
  }, []);

  const handleDelete = async (id) => {
    await axios.delete(`${baseUrl}/cars/${id}`);
    navigate('/', { replace: true });
  };

  return (
    <div className="delete-cars-container">
      {carsList.map((car) => (
        <div key={car.id} className="car-table-card">
          <img src={car.image} alt="car-logo" />
          <h2>{car.name}</h2>
          <button type="button" onClick={() => handleDelete(car.id)}>Delete</button>
        </div>
      ))}
    </div>
  );
};

export default DeleteCar;
