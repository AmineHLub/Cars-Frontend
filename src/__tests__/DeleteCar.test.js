import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import DeleteCar from '../components/Pages/DeleteCar';

describe('Testing Delete car component', () => {
  test('renders a correct data', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router><DeleteCar /></Router>
      </Provider>,
    );
    expect(getByTestId('cars-container')).toBeInTheDocument();
    expect(getByTestId('h1')).toBeInTheDocument();
    expect(getByTestId('p')).toBeInTheDocument();
  });

  test('renders as expected', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router><DeleteCar /></Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
