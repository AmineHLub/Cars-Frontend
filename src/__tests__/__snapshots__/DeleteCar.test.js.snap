// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Delete car component renders as expected 1`] = `
<div
  className="absolute h-screen bg-primaryGreen w-full"
>
  <div
    className="bg-show-car opacity-20"
  />
  <div
    className="text-white opacity"
  />
  <div
    className="absolute top-0 h-screen w-full flex items-start justify-center"
  >
    <div
      className="py-4 px-4 flex flex-col items-start justify-center"
    >
      <div
        className="text-center p-4 w-full"
      >
        <h1
          className="text-white text-lg md:text-2xl md:mb-10 font-bold"
          data-testid="h1"
        >
          DELETE A CAR
        </h1>
        <p
          className="text-white mt-5 md:text-xl"
          data-testid="p"
        >
          Click on the delele button to perform the delele operation.
        </p>
      </div>
      <div
        className="mx-auto p-2 md:mt-10 w-full md:w-full"
        data-testid="cars-container"
      >
        <div
          className="border border-white rounded-md p-2"
        />
      </div>
    </div>
  </div>
</div>
`;
